### PR TITLE
Replaced dropdowns with `VectorSelector`

### PR DESCRIPTION
**Pull request by rubenthoms**
Replaced `dcc.DropDown` with `VectorSelector` for selecting vectors in reservoir simulation timeseries plugin.

### CODE REVIEW

**Review by HansKallekleiv — CHANGES_REQUESTED** [review]
Some comments based from testing the plugin interactively on the Reek testdata.
The selector looks good and intuitive to use! A real improvement compared to the dropdown layout.
I encountered two issues that should be fixed:
- Initial vectors specified in the config file is correctly showing in the `VectorSelector`, but is not rendered in the plotly graph.
- Atleast one vector is rendered as subplots 3 times. 
image

Also, the changes here needs an update to the [example config](https://github.com/equinor/webviz-subsurface-testdata/blob/master/webviz_examples/webviz-full-demo.yml#L140)

File: webviz_subsurface/plugins/_reservoir_simulation_timeseries.py
Comment (anders-kiaer) [inline webviz_subsurface/plugins/_reservoir_simulation_timeseries.py:773]:
```suggestion
                rows=max(len(vectors), 1),
```

File: webviz_subsurface/plugins/_reservoir_simulation_timeseries.py
Comment (anders-kiaer) [inline webviz_subsurface/plugins/_reservoir_simulation_timeseries.py:777]:
```suggestion
                subplot_titles=titles if titles else ["No vector selected"],
```

**Review by anders-kiaer — COMMENTED** [review]
LGTM, I only have a couple of minor comments. Since this touches `AVG_`, `INTVL_` and also a plugin which is used quite a lot, I summon @asnyv for more 👀 before merge.

**Comment by asnyv** [comment]
A wild @asnyv has appeared!

Not sure if you have tested to throw in some csv-input with more or less random column names + checked how so called "UDQs" work? I'll try to do some testing on my side with various inputs to see how things work out. As @anders-kiaer mentions, it is a very frequently used plugin, so wise to spend some time on QC 🧐

File: webviz_subsurface/plugins/_reservoir_simulation_timeseries.py
Comment (asnyv) [inline webviz_subsurface/plugins/_reservoir_simulation_timeseries.py:90]: is this breaking change really necessary? Easy to wrap to existing `vector1`, `vector2`, and `vector3` into a list in the code instead (checking if they are defined of course). I agree that it is cleaner, and probably better, but I think it isn't worth it to break existing setups for this.

File: webviz_subsurface/plugins/_reservoir_simulation_timeseries.py
Comment (asnyv) [inline webviz_subsurface/plugins/_reservoir_simulation_timeseries.py:889]: this extra indent looks funky. Sure this is correct?

File: webviz_subsurface/plugins/_reservoir_simulation_timeseries.py
Comment (rubenthoms) [reply webviz_subsurface/plugins/_reservoir_simulation_timeseries.py:90]: The existing setups are still working and we are showing a deprecation warning. I just wanted to remove it from the docs since it is going to be deprecated, i.e. users new to this plugin should rather stick to the new way of defining vectors. Does that make sense?

File: webviz_subsurface/plugins/_reservoir_simulation_timeseries.py
Comment (rubenthoms) [reply webviz_subsurface/plugins/_reservoir_simulation_timeseries.py:889]: Hehe, I'll better gonna consult my local pylint installation ;)

Comment (rubenthoms) [reply webviz_subsurface/plugins/_reservoir_simulation_timeseries.py]: Good catch!

File: webviz_subsurface/plugins/_reservoir_simulation_timeseries.py
Comment (asnyv) [reply webviz_subsurface/plugins/_reservoir_simulation_timeseries.py:90]: I see that it is deprecated, I just don't think it is worth it to deprecate it (especially considering we are not making other changes on the input side as far as I know). 
I don't see the point of adding the noise with a deprecation warning (and thus possible questions from users). It is slightly cleaner from the code side of course, but don't underestimate the extra issues you introduce on the user side due to the bunch of existing setups. 

We will probably change other parts of the input for this plugin at a later stage, and considering users don't like to do more changes / updates to their setups than absolutely necessary, I suggest that we delay this deprecation to a time when we make a larger change to the plugin's input.

**Comment by rubenthoms** [comment]
@asnyv @anders-kiaer hmm, I have to admit that I found this plugin to be rather hard to read/understand/maintain. I heard that it was written/extended by several people who were new to the project. If this plugin is one of the most frequently used/important ones, it might be worth it spending some time on refactoring the code/code structure to make it better readable/understandable?

File: webviz_subsurface/plugins/_reservoir_simulation_timeseries.py
Comment (rubenthoms) [reply webviz_subsurface/plugins/_reservoir_simulation_timeseries.py:90]: @asnyv I see your point. Well, I am happy with either solution (we would just need to remove the deprecation warning and add back the old doc lines). Your call. What do you think, @anders-kiaer?

**Comment by asnyv** [comment]
@rubenthoms Yes, this plugin has been touched by a lot of hands. It is also among the older ones, so the style of newer plugins probably differentiate a bit. And there are still many features that could/should be included. So I can fully agree that a rewrite would be useful, but before anyone starts a major rewrite, I think we should make a better scope of which features that are needed and how it should look a bit more long term. 

In the meantime: I think as mentioned that we should be careful with changing the user input without good reasons. Better for users to make one (or a few) big changes, than having to frequently update details.

File: webviz_subsurface/plugins/_reservoir_simulation_timeseries.py
Comment (anders-kiaer) [reply webviz_subsurface/plugins/_reservoir_simulation_timeseries.py:90]: Lets revert to current input (`vector{1,2,3}`). I like the suggestion to change to list as input, so created #677 and we can take that one when refactoring the `ReservoirSimulationTimeSeries` plugin.

**Comment by rubenthoms** [comment]
@asnyv @anders-kiaer  Regarding your second point, Asgeir, I created two PRs in both `wcc` (https://github.com/equinor/webviz-core-components/pull/148) and `wsc` (https://github.com/equinor/webviz-subsurface-components/pull/405) which are making the new property `lineBreakAfterTag (boolean)` available (defaults to false).

**Review by anders-kiaer — APPROVED** [review]

**Review by HansKallekleiv — APPROVED** [review]

**Pull request merged**
